File: 2.c
#include <stdio.h>

#define LIMIT 4000000

int main(void){
    int i;
    long fibnum = 0, sum = 0;
    long n[] = {1, 1};

    for(i = 2; fibnum < LIMIT; i++){
        fibnum = n[1] + n[0];
        n[0] = n[1];
        n[1] = fibnum;

        if(fibnum % 2){
            continue;
        }
        sum += fibnum;
    }
    printf("sum = %ld\n", sum);
    return 0;
}

File: 2.py
#!/usr/bin/env python

LIMIT = 4000000

def fib(limit):
    counter = 0
    cache = [1, 2]
    while True:
        if counter == len(cache):
            cache.append(cache[counter - 1] + cache[counter - 2])
        if cache[counter] > limit:
            return
        yield cache[counter]
        counter += 1

if __name__ == '__main__':
    print sum([ x for x in fib(LIMIT) if x % 2 == 0])
